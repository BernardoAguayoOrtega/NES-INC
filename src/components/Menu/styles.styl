@import "../../stylus/variables.styl";

.menu
  transition: .3s;
  position: fixed;
  right: -200px;
  top: 10%;
  background: lightGray;
  height: 80%;
  width: 20%;
  z-index: 11;
  font-family: Font1;

.menu__list
  list-style: none;
  height: 100%;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  flex-direction: column;

.menu__list-item
  font-size: 2rem

.slide-left
  animation: slide-left 0.8s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;

@keyframes slide-left
  0%
    transform: translateX(0);
  100%
    transform: translateX(-200px);

.slide-right
  animation: slide-right 0.8s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;

@keyframes slide-right
  0%
    transform: translateX(0);
  100%
    transform: translateX(200px);

@media screen and (max-width: 761px)
  .menu
    width: 50%;
  .menu__list-item
    font-size: 1.5rem
@media screen and (min-width: 761px)
  .menu
    width: 30%;
@media screen and (min-width: 1200px)
  .menu
    width: 20%;
