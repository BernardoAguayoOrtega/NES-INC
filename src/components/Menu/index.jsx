// import react
import React, { useState } from 'react';
// import styles
import './styles.styl'

// create and export Home page
const Menu = () => {
  // use state
  const [showMenu, setShowMenu] = useState(true);

  return (
    <>
      <i onClick={() => setShowMenu(value => !value)} className="fa fa-angle-down fa-4x" />
      {showMenu && (
        <div className="menu">Hey there</div>
      )}
    </>
  )
}

export default Menu;
