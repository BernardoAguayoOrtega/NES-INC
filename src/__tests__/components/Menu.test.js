// import react
import React from 'react';
// import shallow
import { mount } from 'enzyme';
// import toJson
import toJson from 'enzyme-to-json';
// import app component
import Menu from '../../components/Menu/index.jsx';

describe('<Menu />', () => {
  const wrapper = mount(<Menu />);

  it('is render the Menu?', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('is there i element?', () => {
    expect(wrapper.find('i').exists()).toBe(true);
  });

  it('is there slide right class name?', () => {
    expect(wrapper.find('div').props().className).toBe('menu slide-right');
  });

  it('is there slide left class name?', () => {
    // expect(wrapper.find('div').props)

    wrapper.find('i').simulate('click')

    expect(wrapper.find('div').props().className).toBe('menu slide-left');
  });

  it('matches the snap shot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
