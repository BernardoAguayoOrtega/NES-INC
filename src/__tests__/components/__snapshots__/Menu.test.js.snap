// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> matches the snap shot 1`] = `
<Menu>
  <i
    className="fa fa-angle-down fa-4x"
    onClick={[Function]}
  />
  <div
    className="menu slide-left"
  >
    <ul
      className="menu__list"
    >
      <li
        className="menu__list-item"
      >
        Category
      </li>
      <li
        className="menu__list-item"
      >
        Category
      </li>
    </ul>
  </div>
</Menu>
`;
